regexp/escape.ts

```diff
@@ -1,6 +1,11 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
 // This module is browser compatible.
 
+/**
+ * This module contains functions to escape strings for use in regular expressions.
+ * @module
```

Comment:
We can copy over the example from the docs of only exported function but it might depend on where this data is rendered on JSR? I see that only mod.ts is rendered on the main page

Reply:
Actually it doesn't make sense to mark escape.ts as an entrypoint considering that there is only one function exported and it's also re-exported from mod.ts

Reply:
Ah, true. Because `std/regexp` currently only has one API, setting `regexp.ts` as entrypoint doesn't make much sense..

Reply:
@kt3k I can open a separate PR to remove it if there is no versioning issue.

Reply:
I'll discuss this with Asher tomorrow. Leave it as is for now (Probably maintainers will handle this if necessary). For now I created an issue https://github.com/denoland/deno_std/issues/4802